interfaces/inc/azure_c_pal/refcount.h

```diff
@@ -108,22 +108,9 @@ then in order to see "count" / "counted" this is a good string to paste in Visua
 
 */
 
-#ifndef DEC_RETURN_ZERO
-#error refcount_os.h does not define DEC_RETURN_ZERO
-#endif // !DEC_RETURN_ZERO
-#ifndef INC_REF_VAR
-#error refcount_os.h does not define INC_REF_VAR
-#endif // !INC_REF
-#ifndef DEC_REF_VAR
-#error refcount_os.h does not define DEC_REF_VAR
-#endif // !DEC_REF
-#ifndef INIT_REF_VAR
-#error refcount_os.h does not define INIT_REF_VAR
-#endif // !INIT_REF
-
-#define INC_REF(type, var) INC_REF_VAR(((REFCOUNT_TYPE(type)*)((unsigned char*)var - offsetof(REFCOUNT_TYPE(type), counted)))->count)
-#define DEC_REF(type, var) DEC_REF_VAR(((REFCOUNT_TYPE(type)*)((unsigned char*)var - offsetof(REFCOUNT_TYPE(type), counted)))->count)
-#define INIT_REF(type, var) INIT_REF_VAR(((REFCOUNT_TYPE(type)*)((unsigned char*)var - offsetof(REFCOUNT_TYPE(type), counted)))->count)
+#define INC_REF(type, var) interlocked_increment(&((REFCOUNT_TYPE(type)*)((unsigned char*)var - offsetof(REFCOUNT_TYPE(type), counted)))->count)
+#define DEC_REF(type, var) interlocked_decrement(&((REFCOUNT_TYPE(type)*)((unsigned char*)var - offsetof(REFCOUNT_TYPE(type), counted)))->count)
+#define INIT_REF(type, var) do { ((REFCOUNT_TYPE(type)*)((unsigned char*)var - offsetof(REFCOUNT_TYPE(type), counted)))->count  = 1; }while(0)
```

Comment:
>->count  = 1; [](start = 119, length = 13)

interlocked_exchange

Reply:
Absolutely, clearly, totally, completely 100% agree.

---
In reply to: [460142297](https://github.com/Azure/azure-c-pal/pull/16#discussion_r460142297) [](ancestors = 460142297)